### ISSUE TITLE
Wrong File name in UIDL checking.

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#202]
**Originally by _opionline@yahoo.de_ on 2010-09-02 04:24:02 +0200**

___
I just test the new uidl function and anything goes wrong.
If i check the account which should keep the files on the
server i get the follow snoopdos log
```
SnoopDos logging started on Donnerstag, 02-Sep-10  at 03:40:51

 Coun Process Name    Action      Target Name                             Options  Res. 
 ---- ------------    ------      -----------                             -------  ---- 
 1    YAM.debug       OpenLib     bsdsocket.library                       Ver 2    OK   
 2    YAM.debug       OpenDev     timer.device                            Unit 0   OK   
 3    YAM.debug       FindSem     NSDPatch (8.3.2002)                              OK   
 4    YAM.debug       OpenLib     miami.library                           Ver 10   OK   
 5    YAM.debug       OpenLib     mui/Gauge.mui                           Ver 19   OK   
 6    YAM.debug       OpenLib     mui/Gauge.mui                           Ver 19   OK   
 7    YAM.debug       OpenLib     NListview.mcc                           Ver 0    OK   
 8    YAM.debug       Lock        YAM:.uidl_mcpbugs_pop.mail.yahoo.com    Read     Fail 
 9    YAM.debug       Lock        « 'YAle'e 'YAle 'YAle 'YAle 'YA
ÌZ 
+w Read     Fail 
 10   YAM.debug       LockScreen  Workbench                                        OK   
 11   YAM.debug       OpenLib     muiwinborder.class                      Ver 19   OK   
 12   YAM.debug       GetVar      AmiSSL/SSL_CLIENT_VERSION               Global   Fail 
 13   YAM.debug       GetVar      SSL_CERT_FILE                           Any*     Fail 
 14   YAM.debug       Open        AmiSSL:cert.pem                         Read     Fail 
 15   YAM.debug       GetVar      SSL_CERT_DIR                            Any*     Fail 



 49   YAM.debug       Open        YAM:logs/YAM-Sep2010.log                Modify   OK   
 50   YAM.debug       ChangeDir   DH6:YAM/logs                                          
 51   YAM.debug       ChangeDir   DH6:YAM                                               
 52   YAM.debug       Lock        YAM:.uidl_mcpbugs_pop.mail.yahoo.com    Read     Fail 
 53   YAM.debug       Open        DH6:YAM/ÿÿÿ                            Modify   OK   
 54   YAM.debug       ChangeDir   DH6:YAM                                               
 55   YAM.debug       ChangeDir   DH6:YAM                                               
 56   YAM.debug       Lock        DH6:YAM                                 Read     OK   
```

You see in line 9 and 52 here are wrong file names used.
After that a file 'ÿÿÿ' was created with the same content as the
original .uidl file.

I start a new session and wrote a @all all log.
Here are a part of them:
```
00:C:     YAM_FI.c:1424:Entering FreeRuleSearchData
00:C:     YAM_FI.c:1434:Leaving FreeRuleSearchData
00:C:    YAM_FI.c:1521:Leaving AllocFilterSearch (result 0x00000000, 0)
00:C:    pop3.c:371:Entering TR_ConnectPOP
00:C:     MailServers.c:200:Entering GetMailServer
00:C:     MailServers.c:224:Leaving GetMailServer (result 0x0d172b8c, 219622284)
00:D:     pop3.c:383:connect to POP3 server 'pop.mail.yahoo.com'
00:C:     UIDL.c:101:Entering InitUIDLhash
00:C:      HashTable.c:724:Entering HashTableGetDefaultStringOps
00:C:      HashTable.c:725:Leaving HashTableGetDefaultStringOps (result 0x0d2b2918, 220932376)
00:C:      HashTable.c:415:Entering HashTableNew
00:C:       HashTable.c:454:Entering HashTableInit
00:C:        HashTable.c:233:Entering DefaultHashAllocTable
00:C:        HashTable.c:237:Leaving DefaultHashAllocTable (result 0x0d75ba78, 225819256)
00:C:       HashTable.c:480:Leaving HashTableInit (result 0x00000001, 1)
00:C:      HashTable.c:426:Leaving HashTableNew (result 0x0d1bbd90, 219921808)
00:C:      UIDL.c:49:Entering BuildUIDLFilename
00:C:       YAM_UT.c:1852:Entering CreateFilename
00:C:        YAM_UT.c:1833:Entering AddPath
00:C:        YAM_UT.c:1842:Leaving AddPath (result 0x0d3d9afc, 222141180)
00:C:       YAM_UT.c:1856:Leaving CreateFilename (result 0x0d3d9afc, 222141180)
00:C:      UIDL.c:90:Leaving BuildUIDLFilename (result 0x0d3d9afc, 222141180)
00:C:      FileInfo.c:58:Entering ObtainFileInfo
***00:D:       FileInfo.c:62:getting file information 0 of file 'YAM:.uidl_mcpbugs_pop.mail.yahoo.com'
00:C:      FileInfo.c:206:Leaving ObtainFileInfo (result 0x00000000, 0)
00:C:      UIDL.c:49:Entering BuildUIDLFilename
00:C:       YAM_UT.c:1852:Entering CreateFilename
00:C:        YAM_UT.c:1833:Entering AddPath
00:C:        YAM_UT.c:1842:Leaving AddPath (result 0x0d3d988c, 222140556)
00:C:       YAM_UT.c:1856:Leaving CreateFilename (result 0x0d3d988c, 222140556)
00:C:      UIDL.c:90:Leaving BuildUIDLFilename (result 0x0d3d988c, 222140556)
00:C:      FileInfo.c:58:Entering ObtainFileInfo
***00:D:       FileInfo.c:62:getting file information 0 of file 'YAle 'YAle 'YAle 'YAle 'YAle 'YAle'
00:C:      FileInfo.c:206:Leaving ObtainFileInfo (result 0x00000000, 0)
00:W:      UIDL.c:148:no or empty .uidl file found
00:C:      UIDL.c:155:uidlHash->hash->entryCount = 0, 0x00000000
00:C:     UIDL.c:167:Leaving InitUIDLhash (result 0x0d73c878, 225691768)
00:C:     ImageArea.c:543:Entering m_ImageArea_OM_SET
00:C:     ImageArea.c:627:Leaving m_ImageArea_OM_SET (result 0x00000000, 0)
00:C:     ImageArea.c:543:Entering m_ImageArea_OM_SET
00:C:     ImageArea.c:627:Leaving m_ImageArea_OM_SET (result 0x00000000, 0)
00:C:     YAM_UT.c:4465:Entering Busy
```

I mark the interesting lines with a ***.
After this session not such bad file was created.

BTW: Maybe YAM can move the existing uidl's in the old .uidl
file to the new file if the uidl match?
Than can avoid to load a lot of files again and avoid also
maybe angry users.

Used Version: YAM 2.7-dev [OS3/m68k] (02.09.2010)

**Comment by tboeckel** [comment]
**Originally on 2010-09-02 08:46:55 +0200**

___
Fixed in the next nightly build.

- UIDL.c: the generated UIDL file name is no longer placed on the stack as a
  local variable inside the generating function. This closes #202.

**Comment by tboeckel** [comment]
**Originally on 2010-09-02 13:25:32 +0200**

___
In (8f59e0c):
 * UIDL.c: the generated UIDL file name is no longer placed on the stack as a
  local variable inside the generating function. This closes #202.

**Comment by tboeckel** [comment]
**Originally on 2010-09-02 13:30:26 +0200**

___
In (8f59e0c):
 * UIDL.c: the generated UIDL file name is no longer placed on the stack as a
  local variable inside the generating function. This closes #202.
